### PR TITLE
feat(ras-acc): add modal checkout GA4 events

### PR DESCRIPTION
**Pull request by laurelfulford**
### All Submissions:

* [ ] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Part of https://github.com/Automattic/newspack-plugin/pull/3179 -- see the testing steps there!

### How to test the changes in this Pull Request:

1.
2.
3.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/class-modal-checkout.php [hunk 1]
```diff
@@ -275,6 +277,28 @@ function ( $item ) {
 			\wp_safe_redirect( apply_filters( 'newspack_blocks_checkout_url', $checkout_url ) );
 			exit;
 		}
+
+		$checkout_url = add_query_arg(
+			$query_args,
```
Comment (chickenn00dle) [inline includes/class-modal-checkout.php]: `$query_args` is not defined when the checkout is for a donation, so this is throwing an undefined warning.

File: includes/class-modal-checkout.php [hunk 2]
```diff
@@ -275,6 +277,28 @@ function ( $item ) {
...
+			\wc_get_page_permalink( 'checkout' )
+		);
+
+		// Get data to send for this purchase.
+		$checkout_button_metadata = [
+			'currency'   => \get_woocommerce_currency(),
+			'amount'     => $price,
+			'product_id' => $product_id,
+			'referer'    => $referer,
```
Comment (chickenn00dle) [inline includes/class-modal-checkout.php]: Same issue as `$query_args` above for `$amount` and `$referer`. This is throwing undefined warnings for donation purchases, and as a result donations look to be breaking modal checkout. At least on my end the modal just loads nonstop while data events seemingly retry nonstop:

```
[10-Jul-2024 20:11:21 UTC] PHP Warning:  Undefined variable $query_args in /Users/raz/Sites/newspack/blocks/includes/class-modal-checkout.php on line 282
[10-Jul-2024 20:11:21 UTC] PHP Warning:  Undefined variable $price in /Users/raz/Sites/newspack/blocks/includes/class-modal-checkout.php on line 289
[10-Jul-2024 20:11:21 UTC] PHP Warning:  Undefined variable $referer in /Users/raz/Sites/newspack/blocks/includes/class-modal-checkout.php on line 291
```

File: includes/class-modal-checkout.php [hunk 3]
```diff
@@ -1482,5 +1507,38 @@ public static function maybe_add_checkout_registration_order_meta( $order ) {
...
+
+		// Get the cart subtotal.
+		$amount = WC()->cart->get_displayed_subtotal();
+
+		// Data to send along with this action.
+		$metadata = [
+			'product_id'       => $product_id,
+			'amount'           => $amount,
+			'currency'         => get_woocommerce_currency(),
+			'modal_pagination' => '2',
```
Comment (chickenn00dle) [inline includes/class-modal-checkout.php:1532]: Non-blocking question: Is there a reason we are only adding this meta when continue is clicked? Seems strange not to include this when initializing the modal as well with a value of 1.

**Review by chickenn00dle — CHANGES_REQUESTED** [review]
Nice work @laurelfulford!

Events seem to be working for non-donation products, so I think we are mostly there. 

I did have some issues with donation products though. I left some comments with more details.

**Comment by laurelfulford** [comment]
Thanks @chickenn00dle! I'm seeing the never-ending checkout loading issue, too, though I did have this working locally at one point 😕 I'll work on getting it untangled!

File: includes/class-modal-checkout.php (hunk 3 above)
Comment (laurelfulford) [reply includes/class-modal-checkout.php:1532]: This is something I need @kmwilkerson's help to confirm -- it seems a little strange to me, too, but it's entirely possible I misunderstood the details in the Asana task!

File: includes/class-modal-checkout.php (hunk 1 above)
Comment (laurelfulford) [reply includes/class-modal-checkout.php]: Thanks for catching this, @chickenn00dle! This (and the next bit of feedback with undefined warnings) should be fixed in [0805cc6](https://github.com/Automattic/newspack-blocks/pull/1768/commits/0805cc6a53a4dd1b509deecda65e76b13053b3c1). 

It looks like I ended with some duplicated code in and outside of the `$is_newspack_checkout` check, with some of this ending up outside of it -- I'm betting it came from a bad merge on my part!

**Comment by laurelfulford** [comment]
Closing in favour of this reworked draft with front-end events: https://github.com/Automattic/newspack-blocks/pull/1819

**Pull request closed without merging**